src/qibolab/_core/instruments/qm/controller.py

```diff
@@ -387,15 +395,30 @@ def register_duration_sweeper_pulses(
         https://docs.quantum-machines.co/latest/docs/Guides/features/?h=interpo#dynamic-pulse-duration
         """
         for pulse in sweeper.pulses:
-            if isinstance(pulse, (Align, Delay)):
+            if isinstance(pulse, Align):
                 continue
 
-            params = args.parameters[pulse.id]
             ids = args.sequence.pulse_channels(pulse.id)
+            params = args.parameters[pulse.id]
+            if abs(sweeper.values[1] - sweeper.values[0]) < 1:
```

Comment:
This condition is repeated in the commented instance in `instructions.py`. Isn't there a better way to just let the user configure and then access the sampling rate?

Reply:
Yes, the sampling rate configured by the user is accessible here doing `channel_sampling_rate(configs[ids[0]])` (it is already in line 407 below). So if we decide to replace `doubled_samples` with the sampling rate, I believe it is possible to do here.